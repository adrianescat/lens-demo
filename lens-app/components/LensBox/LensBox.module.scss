.lensBox {
  display: flex;
  border: 1px solid #f0f0f0;
  border-radius: 5px;
  padding: 20px;
  cursor: pointer;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  position: relative;
  margin: 10px;
  height: 250px;
  width: 200px;
  flex-wrap: wrap;
  flex-direction: column;

  &::after {
    content: "";
    position: absolute;
    z-index: -1;
    width: 100%;
    height: 100%;
    opacity: 0;
    top: 0;
    border-radius: 5px;
    box-shadow: 0 5px 15px rgb(0 0 0 / 30%);
    transition: opacity 0.3s ease-in-out;
    left: 0;
  }

  &:hover::after {
    opacity: 1;
  }

  .name {
    font-family: 'MSQTSemiBold';
    font-size: 24px;
    line-height: 1.1;
  }

  .id {
    color: lightgrey;
    font-size: 14px;
  }

  .summaryScore,
  .patientScore {
    padding-top: 10px;
    width: 100%;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }

  .scoreIcon,
  .patientIcon {
    margin-right: 5px;
    width: 20px;
  }

  .scoreIcon {
    color: #efe01d;
  }

  .patientIcon {
    color: #8C03A3;
  }

  .flagIcon {
    color: #fff;
    width: 20px;
    transform: rotate(-45deg);
  }

  .lastUpdated {
    font-size: 11px;
    padding-top: 10px;
  }

  .urgentWrapper {
    width: 85px;
    height: 88px;
    overflow: hidden;
    position: absolute;
    top: -3px;
    right: -3px;

    .urgentRibbon {
      font: bold 15px;
      color: #fff;
      text-align: center;
      text-shadow: rgba(255,255,255,0.5) 0px 1px 0px;
      -webkit-transform: rotate(45deg);
      -moz-transform: rotate(45deg);
      -ms-transform: rotate(45deg);
      -o-transform: rotate(45deg);
      position: relative;
      padding: 7px 0;
      left: 3px;
      top: 3px;
      width: 120px;
      background-color: #F00543;
    }
  }

  .lensIcon {
    position: absolute;
    bottom: 2px;
    right: 10px;
  }
}