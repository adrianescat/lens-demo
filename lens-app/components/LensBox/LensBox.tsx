import React, { ReactElement } from 'react'
import { faStar, faHospitalUser, faFlag } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import dateFormat from 'dateformat';
import Link from 'next/link';
import { Lens } from '../../types/lens';
import styles from './LensBox.module.scss';
import LensIcon from '../LensIcon';

export interface Props {
  data: Lens;
}

export default function LensBox({ data }: Props): ReactElement {
  const date = dateFormat(new Date(data.lastUpdated), 'mm/dd/yyyy');

  return (
    <Link href={`/lens/${data.id}`}>
      <div role="Lens" aria-label="Lens" className={styles.lensBox}>
        <div>
          <div className={styles.name}>
            {`${data.patient.firstName} ${data.patient.lastName}`}
          </div>
          <div className={styles.id}>id: {data.patient.emrId}</div>
        </div>
        <div className={styles.summaryScore} title="Summary score" aria-label="Summary score">
          <span>
            <FontAwesomeIcon className={styles.scoreIcon} icon={faStar} />
          </span>
          <span>
            {data.summaryScore}
          </span>
        </div>

        <div className={styles.patientScore} title="Patient score" aria-label="Patient score">
          <span>
            <FontAwesomeIcon className={styles.patientIcon} icon={faHospitalUser} />
          </span>
          <span>
            {data.patientScore}
          </span>
        </div>
        <div className={styles.lastUpdated}>Last updated: {date}</div>
        {data.urgent && (
          <div className={styles.urgentWrapper} title="Urgent" aria-label="Urgent">
            <div className={styles.urgentRibbon}>
              <FontAwesomeIcon className={styles.flagIcon} icon={faFlag} />
            </div>
          </div>)
        }
        <LensIcon label={data.summaryLabel} title={data.summary} className={styles.lensIcon} />
      </div>
    </Link>
  )
}
