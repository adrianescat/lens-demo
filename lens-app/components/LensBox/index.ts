import LensBox from './LensBox';

export default LensBox;