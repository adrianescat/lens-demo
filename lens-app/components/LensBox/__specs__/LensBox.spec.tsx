import React from 'react';
import { shallow } from 'enzyme';
import LensBox, { Props } from '../LensBox';

let wrapper: any;
let props: Props;

const lens = {
  "id": "5221291239183",
  "patient": {
    "dateOfBirth": "1940-06-12T00:00:00",
    "emrId": "MT00034",
    "firstName": "Michael",
    "homePhone": "650-477-4676",
    "lastName": "Thompson",
    "middleName": "B",
    "mobilePhone": "",
    "workPhone": ""
  },
  "patientScore": 70.97112000000001,
  "summary": "Metabolic Lab Report",
  "summaryLabel": "metabolic_lab_report",
  "summaryScore": 1.3,
  "urgent": false,
  "lastUpdated": "2020-03-12T00:00:00",
  "files": [
    {
      "order": 0,
      "id": "5221291239183_0",
      "width": 1700,
      "height": 2200
    }
  ]
}

beforeEach(() => {
  props = { data: lens };
  wrapper = shallow(<LensBox {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});