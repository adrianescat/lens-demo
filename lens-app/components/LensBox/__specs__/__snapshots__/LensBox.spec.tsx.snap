// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<Link
  href="/lens/5221291239183"
>
  <div
    aria-label="Lens"
    className="lensBox"
    role="Lens"
  >
    <div>
      <div
        className="name"
      >
        Michael Thompson
      </div>
      <div
        className="id"
      >
        id: 
        MT00034
      </div>
    </div>
    <div
      aria-label="Summary score"
      className="summaryScore"
      title="Summary score"
    >
      <span>
        <FontAwesomeIcon
          border={false}
          className="scoreIcon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f005",
                "M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z",
              ],
              "iconName": "star",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <span>
        1.3
      </span>
    </div>
    <div
      aria-label="Patient score"
      className="patientScore"
      title="Patient score"
    >
      <span>
        <FontAwesomeIcon
          border={false}
          className="patientIcon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                640,
                512,
                Array [],
                "f80d",
                "M480 320a96 96 0 1 0-96-96 96 96 0 0 0 96 96zm48 32a22.88 22.88 0 0 0-7.06 1.09 124.76 124.76 0 0 1-81.89 0A22.82 22.82 0 0 0 432 352a112 112 0 0 0-112 112.62c.14 26.26 21.73 47.38 48 47.38h224c26.27 0 47.86-21.12 48-47.38A112 112 0 0 0 528 352zm-198.09 10.45A145.19 145.19 0 0 1 352 344.62V128a32 32 0 0 0-32-32h-32V32a32 32 0 0 0-32-32H96a32 32 0 0 0-32 32v64H32a32 32 0 0 0-32 32v368a16 16 0 0 0 16 16h288.31A78.62 78.62 0 0 1 288 464.79a143.06 143.06 0 0 1 41.91-102.34zM144 404a12 12 0 0 1-12 12H92a12 12 0 0 1-12-12v-40a12 12 0 0 1 12-12h40a12 12 0 0 1 12 12zm0-128a12 12 0 0 1-12 12H92a12 12 0 0 1-12-12v-40a12 12 0 0 1 12-12h40a12 12 0 0 1 12 12zm48-122a6 6 0 0 1-6 6h-20a6 6 0 0 1-6-6v-26h-26a6 6 0 0 1-6-6v-20a6 6 0 0 1 6-6h26V70a6 6 0 0 1 6-6h20a6 6 0 0 1 6 6v26h26a6 6 0 0 1 6 6v20a6 6 0 0 1-6 6h-26zm80 250a12 12 0 0 1-12 12h-40a12 12 0 0 1-12-12v-40a12 12 0 0 1 12-12h40a12 12 0 0 1 12 12zm0-128a12 12 0 0 1-12 12h-40a12 12 0 0 1-12-12v-40a12 12 0 0 1 12-12h40a12 12 0 0 1 12 12z",
              ],
              "iconName": "hospital-user",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <span>
        70.97112000000001
      </span>
    </div>
    <div
      className="lastUpdated"
    >
      Last updated: 
      03/12/2020
    </div>
    <LensIcon
      className="lensIcon"
      label="metabolic_lab_report"
      title="Metabolic Lab Report"
    />
  </div>
</Link>
`;
